async_fuse/src/fuse/fuse_reply.rs

```diff
@@ -229,64 +263,105 @@ impl_fuse_reply_error_for! {
     ReplyXAttr,
 }
 
-#[cfg(target_os = "macos")]
-impl_fuse_reply_error_for! {
-    ReplyXTimes,
+macro_rules! impl_as_iovec_for {
+    {$($t:ty,)+} => {
+        $(impl AsIoVec for $t {
+            #[allow(dead_code)]
+            fn as_io_vec(&self) -> IoVec<&[u8]> {
+                let type_len = mem::size_of::<$t>();
+                let bytes: &[u8] = match type_len {
+                    0 => &[],
+                    l => {
+                        let p = utilities::cast_to_ptr(self);
+                        unsafe { slice::from_raw_parts(p, l) }
+                    }
+                };
```

Comment:
There is a safe wrapper `crate::fuse::abi_marker::as_abi_bytes`.

https://github.com/datenlord/datenlord/blob/28be4f773d78ed3116b8c63f4df88e26fa7f57fc/async_fuse/src/fuse/abi_marker.rs#L25-L30

Reply:
Yes, it works for all FuseAbiData, but there's a tuple in this case, `(FuseEntryOut, FuseOpenOut)`, which is not covered.

Reply:
Tuples do not have any guarantees about their layout. Transmuting them is **UB**.
<https://doc.rust-lang.org/reference/type-layout.html#tuple-layout>
It is impossible to impl `AsIoVec` for `(FuseEntryOut, FuseOpenOut)`.

Reply:
Let me add this tuple to the trait FuseAbiData implementation.

Reply:
No. `(FuseEntryOut, FuseOpenOut)` does not satisfy the safety requirements of `FuseAbiData`.

You may need to write a struct like this:

```rust
#[repr(C)] // layout guarantee
pub struct Pair<T1, T2>{
    first: T1,
    second: T2
}

impl<T1,T2> Pair<T1,T2>{
   pub fn new(first: T1, second: T2) -> Self {
       assert_eq!(align_of::<T1>() == align_of::<T2>()); // Ensure there are no padding bytes.
       ...
   }
}

impl AsIoVec for Pair<FuseEntryOut, FuseOpenOut> {
   ...
}
```